src/wasm/wasm-binary.cpp

```diff
@@ -55,7 +55,7 @@ void WasmBinaryWriter::write() {
   writeFunctions();
   writeDataSegments();
   if (debugInfo) writeNames();
-  if (sourceMap) writeSourceMapUrl();
+  if (sourceMap && !sourceMapUrl.empty()) writeSourceMapUrl();
   if (symbolMap.size() > 0) writeSymbolMap();
 
   if (sourceMap) {
```

Comment:
should this line also check that? or is it valid to not have the URL but have the epilog?

Reply:
That's related to writing of the external file. The writing of writeSourceMapProlog/writeSourceMapEpilog does not depend on presence of sourceMapUrl. I just wanted to change the above to `if (!sourceMapUrl.empty())`, but thought it will be weird to write just URL in the wasm without generated map file.

Reply:
I see, sounds good.